plugins/in_logcat/logprint.c

```diff
@@ -0,0 +1,280 @@
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2015-2022 The Fluent Bit Authors
+ *  Copyright (C) 2005-2017 The Android Open Source Project
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "logprint.h"
+#include <string.h>
+
+int parseLogEntry(log_msg * buf, int length, AndroidLogEntry * entry,
+                  struct flb_input_instance *ins)
+{
+    /* The following code is a modified copy of AOSP's android_log_processLogBuffer */
+
+    if (length < (int) (sizeof(buf->entry))) {
```

Comment:
I'm not sure if this way of comparing the size is the accepted one or if using a constant that's used definition time is (I prefer the latter)